tests/ManageCourses.Tests/TestUtilities/ApiLocalWebHost.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.Extensions.Configuration;
+
+namespace GovUk.Education.ManageCourses.Tests.TestUtilities
+{    
+    public class ApiLocalWebHost
+    {        
+        private readonly IWebHost theHost;
+        private int hasLaunched = 0;
+
+        public ApiLocalWebHost(IConfiguration config)
+        {
+            theHost = WebHost.CreateDefaultBuilder()
+                .UseStartup<ManageCourses.Api.Startup>()
+                .UseConfiguration(config)
+                .Build();                 
+        }
+
+        public ApiLocalWebHost Launch()
+        {
+            bool shouldLaunch = 0 == Interlocked.Exchange(ref hasLaunched, 1);
+
+            if (shouldLaunch)
+            {
+                theHost.RunAsync();
+                Thread.Sleep(5000); //:( 
```

Comment:
Explain the reason this hack has to be there so that if the reason goes away then the code can be removed without having to figure out why it's there first.